app/services/katello/pulp3/repository/yum.rb

```diff
@@ -15,6 +15,7 @@ def remote_options
           options = common_remote_options
           options.merge!(sles_auth_token: sles_token) if sles_token
           options.merge!(url: url, policy: root.download_policy)
+          options.merge!(retain_package_versions: root.retain_package_versions_count) unless root.retain_package_versions_count.nil? || root.retain_package_versions_count.to_i == 0
```

Comment:
This 'unless' causes the whole method to return nil unless you are using  retain_package_versions.  I think this is causing the test failures.

However, i think we always need to add retain_package_versions_count to the remote_options otherwise the user doesn't have the ability to unset that option i don't think?  However i think always setting this will mean you have to re-record a lot of the pulp3 vcr tests :)

Reply:
Thanks a lot!